linkerd/app/admin/src/server.rs

```diff
@@ -57,9 +58,22 @@ impl<M> Admin<M> {
             ready,
             shutdown_tx,
             tracing,
+
+            #[cfg(feature = "pprof")]
+            pprof: None,
         }
     }
 
+    #[cfg(feature = "pprof")]
+    pub fn with_profiling(mut self, enabled: bool) -> Self {
+        self.pprof = if enabled {
+            Some(crate::pprof::Pprof)
+        } else {
+            None
+        };
```

Comment:
nit, feel free to ignore this:
```suggestion
        self.pprof = enabled.then_some(crate::pprof::Pprof);
```